press 'a' to toggle all in multiselect

jeysal opened the pull request:
Resolves #162.
Tested manually because it appears there are no automatic tests for the multiselect prompt.
'a' is disabled (and not shown in instructions) if `maxChoices` are given, because there is no sensible strategy to toggle all if it would exceed the max, so it'd have to behave unintuitively otherwise.

lumio commented:
Thanks for your contribution! This looks really good! I will review it in a moment.

lumio commented on lib/elements/multiselect.js:
```diff
@@ -136,19 +136,35 @@ class MultiselectPrompt extends Prompt {
+      return this.bell();
+    }
+
+    this.value.forEach(v => v.selected = !v.selected);
```
It will always take the opposite state here which might be problematic. If one item is selected, the selected one will be unselected and the rest selected.

Instead of just negate `v.selected` you could do something like this (not tested, just giving an idea here)

```js
const newSelectedState = !this.value[0].selected;
this.value.forEach(v => v.selected = newSelectedState);
```

That way, it always takes the negated state of the first element.

jeysal replied:
The options we have here are:
1. Toggling all (currently implemented by the PR)
2. Set all to the inverse of the first (Your suggestion)
3. Set all to the inverse of the currently "hovered" (Similar to your suggestion)
4. Hold an additional boolean state that cycles between all on and all off regardless of what individual toggles happen in between (e.g. used in yarn upgrade-interactive)

I don't have any strong feelings here, other than 3 IMO being a bit smarter (and not harder to implement) than 2, and 4 introducing extra complexity into the multiselect that I'm not sure is worth it.

Feel free to pick one and I'll follow up when I have time ;)

lumio replied:
Good point and not a bad idea actually. I think however that it doesn't really matter if it's one of 2, 3 or 4 as long as all items are set to the same state 😊

I would say go ahead with 3 then, if it is easy to do.

terkelg commented:
Thank you @jeysal and thanks for reviewing @lumio. Good points!

jeysal commented:
Hey everyone, sorry for taking so long, I'm very short on time atm. I've updated the implementation to the new behavior as discussed and also added special handling for:
* disabled options (are left untouched for 'a')
* pressing 'a' on a disabled option (bell as at the latest the second press there would not do anything)

jeysal commented:
@terkelg do you want to give it a review too? :)

jeysal commented:
@lumio or would your approval be enough to merge it?
